R/pkg/R/SQLContext.R

```diff
@@ -147,6 +147,55 @@ getDefaultSqlSource <- function() {
   l[["spark.sql.sources.default"]]
 }
 
+writeToTempFileInArrow <- function(rdf, numPartitions) {
+  # R API in Arrow is not yet released. CRAN requires to add the package in requireNamespace
+  # at DESCRIPTION. Later, CRAN checks if the package is available or not. Therefore, it works
+  # around by avoiding direct requireNamespace.
+  requireNamespace1 <- requireNamespace
+  if (requireNamespace1("arrow", quietly = TRUE)) {
+    record_batch <- get("record_batch", envir = asNamespace("arrow"), inherits = FALSE)
+    record_batch_stream_writer <- get(
+      "record_batch_stream_writer", envir = asNamespace("arrow"), inherits = FALSE)
+    file_output_stream <- get(
+      "file_output_stream", envir = asNamespace("arrow"), inherits = FALSE)
+    write_record_batch <- get(
+      "write_record_batch", envir = asNamespace("arrow"), inherits = FALSE)
+
+    # Currently arrow requires withr; otherwise, write APIs don't work.
+    # Direct 'require' is not recommended by CRAN. Here's a workaround.
+    require1 <- require
+    if (require1("withr", quietly = TRUE)) {
+      numPartitions <- if (!is.null(numPartitions)) {
+        numToInt(numPartitions)
+      } else {
+        1
+      }
+      fileName <- tempfile(pattern = "spark-arrow", fileext = ".tmp")
+      chunk <- as.integer(ceiling(nrow(rdf) / numPartitions))
+      rdf_slices <- split(rdf, rep(1:ceiling(nrow(rdf) / chunk), each = chunk)[1:nrow(rdf)])
```

Comment:
`1 : ceiling`? `1 : nrow`?

Reply:
how is this slices computed? is it similar to https://github.com/apache/spark/blob/c3b4a94a91d66c172cf332321d3a78dba29ef8f0/R/pkg/R/context.R#L152

Reply:
This resembles PySpark side logic:

https://github.com/apache/spark/blob/d367bdcf521f564d2d7066257200be26b27ea926/python/pyspark/sql/session.py#L554-L556

Let me check the difference between them

Reply:
Let me try to reuse the R side slicing logic.

Reply:
ah, any idea that was done that way in python? this seems to be different from sc.paralleize which is what https://github.com/apache/spark/blob/c3b4a94a91d66c172cf332321d3a78dba29ef8f0/R/pkg/R/context.R#L152 is done

Reply:
Not sure. I think the intention is the same. Let me stick to R's one for now.